### PR TITLE
[ci][release] Add mac 3.11 wheels to release scripts

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
Add 3.11 for mac wheels in
- wheels download from CI 
- wheels check from pypi 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/util/download_wheels.sh [hunk 1]
```diff
@@ -35,17 +35,19 @@ download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERS
 
 # macOS.
 echo "Downloading Ray core MacOS wheels (x86_64)"
-download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERSION/$RAY_HASH/ray-$RAY_VERSION-cp37-cp37m-macosx_10_15_x86_64.whl"
+download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERSION/$RAY_HASH/ray-$RAY_VERSION-cp37-cp37-macosx_10_15_x86_64.whl"
```
Comment (rickyyx) [inline release/util/download_wheels.sh:38]: will test it manually once mac wheels are built on 2.5.1

Comment (aslonnie) [reply release/util/download_wheels.sh:38]: noob question: what does the change from `cp37m` to `cp37` mean?

Comment (rickyyx) [reply release/util/download_wheels.sh:38]: we were generating wheels that had "m" in the wheel name (using the system installed python) before this PR https://github.com/ray-project/ray/pull/36185
And we no longer generate wheels with "m" when we use conda. 

But I am also not sure why "m" in the first place.

File: release/util/download_wheels.sh [hunk 2]
```diff
@@ -35,17 +35,19 @@ download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERS
...
 download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERSION/$RAY_HASH/ray-$RAY_VERSION-cp38-cp38-macosx_10_15_x86_64.whl"
 download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERSION/$RAY_HASH/ray-$RAY_VERSION-cp39-cp39-macosx_10_15_x86_64.whl"
-download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERSION/$RAY_HASH/ray-$RAY_VERSION-cp310-cp310-macosx_10_15_universal2.whl"
+download_wheel "https://s3-us-west-2.amazonaws.com/ray-wheels/releases/$RAY_VERSION/$RAY_HASH/ray-$RAY_VERSION-cp310-cp310-macosx_10_15_x86_64.whl"
```
Comment (can-anyscale) [inline release/util/download_wheels.sh:41]: oh, has this always been broken for 3.10 then?

**Review by can-anyscale — APPROVED** [review]

File: release/util/download_wheels.sh (hunk 2 above)
Comment (rickyyx) [reply release/util/download_wheels.sh:41]: no it's only changed after the PR https://github.com/ray-project/ray/pull/36185

**Review by krfricke — APPROVED** [review]

**Pull request merged**